proteuslib/unit_models/reverse_osmosis_1D.py

```diff
@@ -1021,9 +1022,28 @@ def initialize(blk,
                      initialization (default = None, use default solver)
             optarg : solver options dictionary object (default=None, use default solver options)
 
+            fail_on_warning : boolean argument to fail or only produce  warning upon unsuccessful solve (default=False)
+
         Returns:
             None
         """
+        def _check_solve(results, checkpoint=None):
```

Comment:
This looks like it should be a general utility method so that you do not need to re-implement it in each property package.

Reply:
I think that would be good. Right now we only have a "Tools" directory, which contains the parameter_sweep tool. Do you think we should have a separate "util" folder, like IDAES, or could this live in "Tools"? 

Better yet--this might just belong in `idaes.core.util.initialization`, assuming this isn't redundant compared with what's already there. Should we reserve this for a subsequent PR (either here or on IDAES)?

Reply:
I was suggesting a separate PR to IDAES (as you can't combine it with this one).